src/restconfig/src/test/java/org/geoserver/rest/resources/ResourceTest.java

```diff
@@ -0,0 +1,310 @@
+/* (c) 2016 Open Source Geospatial Foundation - all rights reserved
+ * This code is licensed under the GPL 2.0 license, available at the root
+ * application directory.
+ */
+package org.geoserver.rest.resources;
+
+import java.io.IOException;
+import java.io.InputStream;
+import java.io.OutputStreamWriter;
+import java.text.DateFormat;
+import java.text.SimpleDateFormat;
+import java.util.Date;
+import java.util.HashMap;
+import java.util.Map;
+import java.util.TimeZone;
+
+import net.sf.json.JSON;
+import net.sf.json.JSONObject;
+import net.sf.json.test.JSONAssert;
+
+import org.custommonkey.xmlunit.NamespaceContext;
+import org.custommonkey.xmlunit.SimpleNamespaceContext;
+import org.custommonkey.xmlunit.XMLAssert;
+import org.custommonkey.xmlunit.XMLUnit;
+import org.geoserver.platform.resource.Resource;
+import org.geoserver.platform.resource.Resources;
+import org.geoserver.rest.util.IOUtils;
+import org.geoserver.test.GeoServerSystemTestSupport;
+import org.junit.Assert;
+import org.junit.Before;
+import org.junit.Test;
+import org.w3c.dom.Document;
+
+import com.mockrunner.mock.web.MockHttpServletResponse;
+
+/**
+ * 
+ * @author Niels Charlier
+ *
+ */
+public class ResourceTest extends GeoServerSystemTestSupport {
+    
+    private final NamespaceContext NS_XML, NS_HTML;
+    private final DateFormat FORMAT = new SimpleDateFormat("yyyy-MM-dd HH:mm:ss.S z");
+    private final DateFormat FORMAT_HEADER = new SimpleDateFormat("E, dd MMM yyyy HH:mm:ss z");
+    
+    private Resource myRes; 
+    
+    public ResourceTest() {
+        FORMAT.setTimeZone(TimeZone.getTimeZone("UTC"));
+        FORMAT_HEADER.setTimeZone(TimeZone.getTimeZone("GMT"));
+        Map<String, String> mapXML = new HashMap<String, String>();
+        mapXML.put("atom", "http://www.w3.org/2005/Atom");
+        NS_XML = new SimpleNamespaceContext(mapXML);
+        Map<String, String> mapHTML = new HashMap<String, String>();        
+        mapHTML.put("x", "http://www.w3.org/1999/xhtml");
+        NS_HTML = new SimpleNamespaceContext(mapHTML);        
+    }          
+    
+    @Before
+    public void initialise() throws IOException {                
+        myRes = getDataDirectory().get("/mydir/myres");
+        try (OutputStreamWriter os = new OutputStreamWriter(myRes.out())) {
+            os.append("This is my test.");
+        }
+        
+        try (OutputStreamWriter os = new OutputStreamWriter(getDataDirectory().get("/mydir2/fake.png").out())) {
+            os.append("This is not a real png file.");
+        }
+
+        IOUtils.copyStream(getClass().getResourceAsStream("testimage.png"),
+                getDataDirectory().get("/mydir2/imagewithoutextension").out(), true, true);
```

Comment:
@NielsCharlier please include non-ASCII characters in paths and content. This will give you unit test coverage of charset handling. You can also run Maven tests with -Dfile.encoding as this is used to set the JVM default charset and many parsers will choose it if a charset is not specified.